rotkehlchen/inquirer.py

```diff
@@ -167,52 +169,65 @@ def get_underlying_asset_price(token: EvmToken) -> Optional[Price]:
         price = Inquirer().find_uniswap_v2_lp_price(token)
+        oracle = CurrentPriceOracle.UNISWAPV2
     elif token.protocol == CURVE_POOL_PROTOCOL:
         price = Inquirer().find_curve_pool_price(token)
+        oracle = CurrentPriceOracle.BLOCKCHAIN
     elif token.protocol == YEARN_VAULTS_V2_PROTOCOL:
         price = Inquirer().find_yearn_price(token)
+        oracle = CurrentPriceOracle.BLOCKCHAIN
 
     if token == A_YV1_ALINK:
-        price = Inquirer().find_usd_price(A_ALINK_V1)
+        price, oracle = Inquirer().find_usd_price_and_oracles(A_ALINK_V1)
     elif token == A_YV1_GUSD:
-        price = Inquirer().find_usd_price(A_GUSD)
+        price, oracle = Inquirer().find_usd_price_and_oracles(A_GUSD)
     elif token in (A_YV1_DAI, A_FARM_DAI):
-        price = Inquirer().find_usd_price(A_DAI)
+        price, oracle = Inquirer().find_usd_price_and_oracles(A_DAI)
     elif token in (A_FARM_WETH, A_YV1_WETH):
-        price = Inquirer().find_usd_price(A_ETH)
+        price, oracle = Inquirer().find_usd_price_and_oracles(A_ETH)
     elif token == A_YV1_YFI:
-        price = Inquirer().find_usd_price(A_YFI)
+        price, oracle = Inquirer().find_usd_price_and_oracles(A_YFI)
     elif token in (A_FARM_USDT, A_YV1_USDT):
-        price = Inquirer().find_usd_price(A_USDT)
+        price, oracle = Inquirer().find_usd_price_and_oracles(A_USDT)
     elif token in (A_FARM_USDC, A_YV1_USDC):
-        price = Inquirer().find_usd_price(A_USDC)
+        price, oracle = Inquirer().find_usd_price_and_oracles(A_USDC)
     elif token in (A_FARM_TUSD, A_YV1_TUSD):
-        price = Inquirer().find_usd_price(A_TUSD)
+        price, oracle = Inquirer().find_usd_price_and_oracles(A_TUSD)
     elif token in ASSETS_UNDERLYING_BTC:
-        price = Inquirer().find_usd_price(A_BTC)
+        price, oracle = Inquirer().find_usd_price_and_oracles(A_BTC)
 
     # At this point we have to return the price if it's not None. If we don't do this and got
     # a price for a token that has underlying assets, the code will enter the if statement after
     # this block and the value for price will change becoming incorrect.
     if price is not None:
-        return price
+        return price, oracle
 
     custom_token = GlobalDBHandler().get_evm_token(
         address=token.evm_address,
         chain=ChainID.ETHEREUM,
     )
     if custom_token and custom_token.underlying_tokens is not None:
         usd_price = ZERO
+        oracles = []
         for underlying_token in custom_token.underlying_tokens:
             token = EvmToken(underlying_token.get_identifier(parent_chain=custom_token.chain))
-            usd_price += Inquirer().find_usd_price(token) * underlying_token.weight
+            underlying_asset_price, oracle = Inquirer().find_usd_price_and_oracles(token)
+            usd_price += underlying_asset_price * underlying_token.weight
+            oracles.append(oracle)
+
+        # Set as oracle the most common oracle of the underlying assets
```

Comment:
this is an overkill. Why not just keep the oracle you had from outside the if.

Reply:
Because in this case different oracles were queried

Reply:
I said this is an overkill.

This information is not needed. Can you think of a situation where this is actually needed? Please describe it to me and explain why I would want such detail.

Just set price oracle blockchain and be done with it.
Then all this extra logic of appending the oracle and then doing a `max()` can go away.